packages/jest-snapshot/src/index.ts

```diff
@@ -278,7 +278,23 @@ const _toMatchSnapshot = (config: MatchSnapshotConfig) => {
     config;
   let {received} = config;
 
-  context.dontThrow && context.dontThrow();
+  /** If a test was ran with `test.failing`. Passed by Jest Circus. */
+  const {testFailing = false} = context;
+
+  /**
+   * Run the snapshot matcher as a normal throwing matcher with
+   * no side effects and no error supression.
+   *
+   * Skipped side effects occur in {@link snapshotState.match}
+   * and include updating the snapshot and recording passed/failed
+   * snapshots in the snapshot state.
+   */
+  const pure = testFailing;
```

Comment:
why are we renaming this to `pure`? Makes grepping harder than it needs to without reason (that I can see?)

Reply:
why can't it be `testFailing` all the way through?

Reply:
That's a fair point and it certainly can. I understand where you're coming from, and if you feel that `testFailing` is better to use now, I'll use it instead.

The idea was to separate the matcher's operational state from the trigger of that state, even though when that state only has one trigger, like right now, it does look a bit silly. 

I can imagine that later we can add a feature to utilize pure matches when the runner isn't expected to update the snapshots (no `-u`) which would be another trigger for the same state.

Reply:
I did both. The current commit uses `testFailing` all the way through, and the one before uses `matchOnly`. If I've convinced you, feel free to revert :)

Reply:
>The idea was to separate the matcher's operational state from the trigger of that state

I totally get that point, but I do think it's easier to evaluate good naming when the use case is actually in front of us instead of hypothesising.

Your example sounds like how `--bail` should work, for instance.

---

Thanks for the discussion! 😀